xbmc/peripherals/addons/AddonButtonMapping.cpp

```diff
@@ -28,11 +28,12 @@ using namespace JOYSTICK;
 using namespace PERIPHERALS;
 
 #ifndef SAFE_DELETE
-  #define SATE_DELETE(x)  do { delete (x); (x) = NULL; } while (0)
+  #define SATE_DELETE(x)  do { delete (x); (x) = nullptr; } while (0)
```

Comment:
The memory management is flawed but not broken. Smart pointers should be used instead. will fix and PR when i get a chance